src/equality.rs

```diff
@@ -0,0 +1,139 @@
+use num::Float;
+use approx::ApproxEq;
+
+use {Xyz, Yxy, Lab, Lch, Rgb, Hsl, Hsv , Luma, LabHue, RgbHue, flt};
+
+macro_rules! impl_eq {
+    (  $self_ty: ident , [$($element: ident),+]) => {
+        impl<T: Float + ApproxEq> ApproxEq for $self_ty<T>
+        where T::Epsilon: Copy
+        {
+            type Epsilon = <T as ApproxEq>::Epsilon;
+
+            fn default_epsilon() -> Self::Epsilon {
+                T::default_epsilon()
+            }
+            fn default_max_relative() -> Self::Epsilon {
+                T::default_max_relative()
+            }
+            fn default_max_ulps() -> u32 {
+                T::default_max_ulps()
+            }
+            fn relative_eq(&self, other: &Self, epsilon: Self::Epsilon, max_relative: Self::Epsilon) -> bool {
+                $( T::relative_eq(&self.$element, &other.$element, epsilon, max_relative) )&&+
```

Comment:
Wouldn't it be better to change this to `self.$element.relative_eq(&other.$element, epsilon, max_relative)`? That would let us have only one macro for implementation, instead of special casing hue based types.